solcx/install.py

```diff
@@ -305,49 +324,46 @@ def _download_solc(url, show_progress):
     return content
 
 
+def _chmod_plus_x(executable_path):
+    executable_path.chmod(executable_path.stat().st_mode | stat.S_IEXEC)
+
+
 def _install_solc_linux(version, show_progress, solcx_binary_path=None):
     download = DOWNLOAD_BASE.format(version, "solc-static-linux")
-    binary_path = _check_for_installed_version(version, solcx_binary_path=solcx_binary_path)
-    if binary_path:
-        LOGGER.info(f"Downloading solc {version} from {download}")
-        content = _download_solc(download, show_progress)
-        with open(binary_path, "wb") as fp:
-            fp.write(content)
-        _chmod_plus_x(binary_path)
+    LOGGER.info(f"Downloading solc {version} from {download}")
+    content = _download_solc(download, show_progress)
+    version_location = get_version_location(version, solcx_binary_path)
+    with open(version_location, "wb") as fp:
+        fp.write(content)
+    _chmod_plus_x(version_location)
 
 
 def _install_solc_windows(version, show_progress, solcx_binary_path=None):
     download = DOWNLOAD_BASE.format(version, "solidity-windows.zip")
-    install_folder = _check_for_installed_version(version)
```

Comment:
This line was important - on windows systems each installation has it's own subdirectory.  Because you instead target `solc_binary_path` as the install path, but never set it as not-`None`, all windows installations are failing.

Reply:
Gotcha.  Damn Windows.

Reply:
Master branch has this:
```python3
def _install_solc_windows(version, show_progress, solcx_binary_path=None):
    download = DOWNLOAD_BASE.format(version, "solidity-windows.zip")
    install_folder = _check_for_installed_version(version)
    if install_folder:
        temp_path = _get_temp_folder()
        content = _download_solc(download, show_progress)
        with zipfile.ZipFile(BytesIO(content)) as zf:
            zf.extractall(str(temp_path))
        install_folder = get_solc_folder(solcx_binary_path=solcx_binary_path).joinpath(
            "solc-" + version
        )
        temp_path.rename(install_folder)
```
Is the `if install_folder:` needed?

Reply:
Yes, on master `_check_for_installed_version` is a bit wierd, if it _doesn't_ find an installed version it returns the install path. On windows, this is a directory not a filename.  The issue you're having is that without replacing that check, the install path is always set as `None`.